### DISCUSSION TITLE
Is it possible to extend the ace serve --watch command

### DISCUSSION THREAD

**Discussion by bjornarhagen** [adonisjs/core#3350] [Help]
Hey can I extend or make use of the `ace serve` command somehow so that it runs my bundle command every time there is a file change? Or do I have to use nodemon/watchman or similar?

```ts
"scripts": {
  "dev": "node ace serve --watch",
  "bundle:css": "esbuild --bundle ./resources/styles/styles.css --outfile=./public/styles2.css",
},
```

**Comment by thetutlage** [comment]
Not right now and you have to rely on an external process + watcher for it. But I do imagine adding support for it sometime later. I just have to finalize the API I want to use. Maybe define tasks within the `.adonisrc.json` file or allow hooking in using the JavaScript API.

The JSON version can look as follows.

```json
{
  "triggers": {
    "source:change": "RUN SOME COMMAND",
    "metafile:change": {
      "test": "resources/styles/**/*.css",
      "run": "esbuild --bundle REST OF THE COMMAND"
    }
  }
}
```

And then maybe define some environment variables and share them with the process. Like currently changed file and so on
